src/test/java/edu/unc/lib/boxc/migration/cdm/services/PermissionsServiceTest.java

```diff
@@ -356,6 +357,82 @@ public void loadPermissionMappingsTest() throws Exception {
         assertEquals("none", testMapping.getAuthenticated());
     }
 
+    @Test
+    public void setPermissionExistingEntryTest() throws Exception {
+        writeCsv(mappingBody("default,canViewMetadata,canViewMetadata", "25,none,none", "26,none,none", "27,none,none"));
+        service.capturePreviousState();
+        testHelper.indexExportData("mini_gilmer");
+        Path permissionsMappingPath = project.getPermissionsPath();
+        var options = new PermissionMappingOptions();
+        options.setCdmId("25");
+        options.setEveryone(UserRole.canViewMetadata);
+        options.setAuthenticated(UserRole.canViewMetadata);
+
+        service.setPermissions(options);
+        assertTrue(Files.exists(permissionsMappingPath));
+
+        List<CSVRecord> rows = listCsvRecords(permissionsMappingPath);
+        assertIterableEquals(Arrays.asList("default", "canViewMetadata", "canViewMetadata"), rows.get(0));
+        assertIterableEquals(Arrays.asList("25", "canViewMetadata", "canViewMetadata"), rows.get(1));
+        assertIterableEquals(Arrays.asList("26", "none", "none"), rows.get(2));
+        assertIterableEquals(Arrays.asList("27", "none", "none"), rows.get(3));
+    }
+
+    @Test
+    public void setPermissionNewEntryTest() throws Exception {
+        writeCsv(mappingBody("default,canViewMetadata,canViewMetadata", "25,none,none", "26,none,none"));
+        service.capturePreviousState();
+        testHelper.indexExportData("mini_gilmer");
+        Path permissionsMappingPath = project.getPermissionsPath();
+        var options = new PermissionMappingOptions();
+        options.setCdmId("27");
+        options.setEveryone(UserRole.canViewMetadata);
+        options.setAuthenticated(UserRole.canViewMetadata);
+
+        service.setPermissions(options);
+        assertTrue(Files.exists(permissionsMappingPath));
+
+        List<CSVRecord> rows = listCsvRecords(permissionsMappingPath);
+        assertIterableEquals(Arrays.asList("default", "canViewMetadata", "canViewMetadata"), rows.get(0));
+        assertIterableEquals(Arrays.asList("25", "none", "none"), rows.get(1));
+        assertIterableEquals(Arrays.asList("26", "none", "none"), rows.get(2));
+        assertIterableEquals(Arrays.asList("27", "canViewMetadata", "canViewMetadata"), rows.get(3));
+    }
+
+    @Test
+    public void setPermissionInvalidIdTest() throws Exception {
+        testHelper.indexExportData("mini_gilmer");
```

Comment:
I'd suggest creating an existing CSV file in this test, so that the project is in a valid state for running the set command and won't be sensitive to the order in which validation happens